app/domain/README.md

```diff
@@ -0,0 +1,296 @@
+how we'd do the same thing in vanilla ruby: 
+
+```ruby
+class DoSomeUseCase
+  def initialize(dependency1: dep1, dependency2: dep2)
+    @dependency1 = dep1
+    @dependency2 = dep2
+  end
+
+  def call(arg1, arg2)
+    do_action1(arg1, arg2)
+    do_action2(arg1)
+  end
+
+  private
+
+  def do_action1(arg1, arg2)
+    do_sub_action1(arg1, arg2)
+    do_sub_action2(arg1, arg2)
+  end
+
+  def do_sub_action1(arg1, arg2)
+    @dependency1.something(arg1)
+    @dependency2.something(arg2)
+  end
+
+  def do_sub_action2(arg1, arg2)
+    @dependency2.something(arg2)
+  end
+
+  def do_action2(arg1)
+    # something here
+  end
+end
+```
+
+This is a perfectly legitimate, vanilla ruby alternative to the `CommandClass`,
+with all the same high-level benefits.  
+
+Visually, however, it's a bit noisy: we manually pass the runtime arguments of
+`call` down to every private method, creating clutter.
+
+Our first instinct might be to save the arguments as member variables, giving
+all methods access to them.  But if we did something like:
+
+```ruby
+def call(arg1, arg2)
+  @arg1 = arg1
+  @arg2 = arg2
+  do_action1
+  do_action2
+end
+```
+
+Then our `call` method would no longer be thread safe.  Remember a single
+instance of `DoSomeUseCase` should be able to serve multiple requests
+simultaneously.
+
+So if we want to avoid argument passing, we must introduce a separate object
+that has _both_ the runtime arguments and the original dependencies as
+attributes, and create a new instance of _that_ object every time
+`DoSomeUseCase#call` is invoked:
+
+```ruby
+class DoSomeUseCase
+
+  class Run
+    def initialize(dep1, dep2, arg1, arg2)
+      @dep1 = dep1
+      @dep2 = dep2
+
+      @arg1 = arg1
+      @arg2 = arg2
+    end
+
+    def call
+      do_action1
+      do_action2
+    end
+
+    private
+
+    def do_action1
+      do_sub_action1
+      do_sub_action2
+    end
+
+    def do_sub_action1
+      @dep1.something(@arg1)
+      @dep2.something(@arg2)
+    end
+
+    def do_sub_action2
+      @dep2.something(@arg2)
+    end
+
+    def do_action2
+      # something here
+    end
+  end
+
+  def initialize(dependency1: dep1, dependency2: dep2)
+    @dependency1 = dep1
+    @dependency2 = dep2
+  end
+
+  def call(arg1, arg2)
+    Run.new(dependency1, dependency2, arg1, arg2).call
+  end
+end
+```
+
+We've eliminated the argument-passing clutter and we're still thread safe.
+_But_ the price is boilerplate and complexity: An additional nested class to
+write and keep track of, and the duplication of the dependencies as attributes
+in both `DoSomeUseCase` and the nested `Run` class.
+
+So we now have two options for vanilla ruby alternatives to the `CommandClass`:
+
+1. One which is coneptually simple but cluttered by passing runtime arguments
+   between all the methods.
+1. One which avoids that clutter but at the expense of other complexity and
+   boilerplate.
+
+The `CommandClass` is simply a third option, which takes the second approach --
```

Comment:
Would it be difficult to include an example of code that you would _not_ convert to a command class?